philips_hue.py

```diff
@@ -14,26 +14,90 @@
 
 # Philips Hue API Documentation: https://developers.meethue.com/develop/hue-api/
 
+"""Module to handle interactions with a Philips Hue system.
+
+This module defines Philips Hue classes, as well as
+callback functions that interact with a Philips Hue client.
+"""
+
 import os
 import requests
 import json
 
 
-class PhilipsHueClient(object):
-    def __init__(self, api_url):
-        self.api_url = api_url
+class PhilipsHueClient(object):    
```

Comment:
So I started implementing the philips hue suggestions you made on my code review and I had some questions about the best way of going about this. I realized that in order to set_color on a light, I still need the ip address and username to make the API call to Philips Hue, but this would mean that I would need to pass the ip address and username as attributes to Light (and probably also HueLights). I feel like the concept of a Light and HueLights shouldn't depend on this information though, but maybe there isn't really a way around that. Does the way I'm approaching this so far seem to make sense?